### PR TITLE
show /home based on survey status from correct, show content in /prog…

### PR DESCRIPTION
**Pull request by eilivjl**
…ress based on statuses from userStatuses

Use new structure of statuses in both firebase and in the app to display data on pages /home and /progress based on the progress of the respondent

TODO: add updateUserStatus (from another branch) to /progress to be able to set surveyStatus to COMPLETE in firebase

### CODE REVIEW

File: src/pages/progress.tsx [hunk 1]
```diff
@@ -49,7 +35,7 @@ const Progress = () => {
                         </div>
                     </div>
                 </div>
-                {purchases && purchases.length > 0 && userInfo.diaryStatus &&
+                {userStatuses && userStatuses.journalStatus === StatusConstants.COMPLETE &&
```
Comment (prabu-ssb) [inline src/pages/progress.tsx:38]: How about collapsing all these similar status checks into a utility function which does this --

```
export const isUserStatusComplete = (statusKey: UserStatusKeys): boolean => {
  const {userStatuses} = useContext(UserContext);
  return userStatuses[statusKey] === StatusConstants.COMPLETE;
}
```
sprinkle a few boundary checks in, of course!

Comment (prabu-ssb) [reply src/pages/progress.tsx:38]: then the JSX could be expressed simply as

```
<>
  {isUserStatusComplete(UserStatusKeys.JOURNAL_STATUS) &&
  ...<jsx>
  }
  {!isUserStatusComplete(UserStatusKeys.JOURNAL_STATUS) &&
  ...<jsx>
  }
  {isUserStatusComplete(UserStatusKeys.QUESTIONNAIRE_STATUS) &&
  ...<jsx>
  }
  ...
</>
```

File: src/pages/home.tsx [hunk 2]
```diff
@@ -50,7 +51,7 @@ const Home = () => {
 
     return (
         <Workspace showFooter={true}>
-            {!diaryStatus &&
+            {(!userStatuses ||  userStatuses.surveyStatus !== StatusConstants.COMPLETE) &&
```
Comment (prabu-ssb) [inline src/pages/home.tsx:54]: see comment regarding `isUserStatusComplete(...)`  below

File: src/pages/progress.tsx [hunk 3]
```diff
@@ -1,39 +1,25 @@
 import Workspace from '../components/layout/workspace/Workspace';
 import PageTitle from '../components/common/PageTitle';
-import React, {useContext, useEffect, useState} from 'react';
-import {FireContext, UserContext} from '../contexts';
-import { useTranslation } from 'react-i18next';
+import React, {useContext, useState} from 'react';
+import {FireContext, UserContext, UserStatusesKeys} from '../contexts';
```
Comment (prabu-ssb) [inline src/pages/progress.tsx:4]: I think we should move the UserStatusKeys to a different place ? Maybe `model/User.ts` ? what do you think ?

**Comment by eilivjl** [comment]
Added utility functions to check user statuses in /home and /progress, exchanged update method in useQuestionnaire with Prabus new updateUserStatuses. Update both survey status and journal status on complete. Moved UserStatusKeys to model/user

File: src/firebase/model/User.ts [hunk 4]
```diff
@@ -60,3 +67,19 @@ export const getRespondentDetailsSecure = (respondentDetails: RespondentDetails)
         journalEnd: diaryEnd ? new Date(diaryEnd) : new Date(),
     };
 };
+
+const isStatus = (statusKey: UserStatusesKeys, statusValue: StatusConstants): boolean => {
```
Comment (prabu-ssb) [inline src/firebase/model/User.ts:71]: When I see this together in one place, I think we can do still better - as it is not clear which status we are talking about (given we have statuses on almost everything!). how about --

 ```
const checkProgress = (statusKey: UserStatusesKeys, statusValue: StatusConstants) => {
  const {userStatuses} = useContext(UserContext);
  return userStatuses && userStatuses[statusKey] === statusValue
};

const isProgressComplete = (statusKey) => 
  checkProgress(statusKey, StatusConstants.COMPLETE);

const isProgressStarted = (statusKey) => 
  checkProgress(statusKey, StatusConstants.STARTED);

const isProgressNotStarted = (statusKey) =>
  !(isProgressStarted(statusKey) || isProgressComplete(statusKey));


/* === Journal === */
const isJournalComplete = () => 
  isProgressComplete(UserStatusesKeys.JOURNAL_STATUS);

const isJournalStarted = () => 
  isProgressStarted(UserStatusesKeys.JOURNAL_STATUS);

const isJournalNotStarted = () => 
  isProgressNotStarted(UserStatusesKeys.JOURNAL_STATUS);
```

Comment (prabu-ssb) [reply src/firebase/model/User.ts:71]: will have to check if the use of `useContext` is correct in `checkProgress`. react might find it problematic to call from a simple function. 

do test!

Comment (eilivjl) [reply src/firebase/model/User.ts:71]: Agree to some degree. Maybe my naming convention isnt the best, but I think it is fairly easy to see what status we are talking about based on the key sent into the function. However, this is more readable


Will check context

File: src/hocs/useQuestionnaire.tsx [hunk 5]
```diff
@@ -86,7 +86,7 @@ const useQuestionnaire = () => {
         // ideally it should suffice setting status at one place
         // - including update at both places for the sake of backward compatibility
         const statusPromises = [
-            questionnaireRef.set({status}, {merge: true}),
+            updateUserStatus(UserStatusesKeys.QUESTIONNAIRE_STATUS, status),
             firestore.doc(getStatusesPathForUser(respondentDetails.respondentId))
```
Comment (prabu-ssb) [inline src/hocs/useQuestionnaire.tsx:90]: Aren't we ready to drop this altogether ?
Does'nt `updateUserStatus` do exactly what `firestore.doc().set()` does ?

**Review by prabu-ssb — APPROVED** [review]
Just check if the updateUserStatus and firestore.set is redundant/duplicate ..

looks very good otherwise! :)

File: src/hocs/useQuestionnaire.tsx (hunk 5 above)
Comment (eilivjl) [reply src/hocs/useQuestionnaire.tsx:90]: loks like I deleted the worng thing. The top one there is for setting questionnaire-status at /questionnaire/data ?

**Review by prabu-ssb — APPROVED** [review]
nice! :)

**Pull request merged**